### PR TITLE
Add pitch and volume to audio in animations

### LINKED ISSUE #91605: AudioStreamPlayer pitch unnafected by AnimationPlayer
### Tested versions

Reproducible in 4.2.1.stable (haven't had time to reproduce in other versions, sorry. If needed I can do in a couple of days)

### System information

Godot v4.2.1.stable - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1080 Ti (NVIDIA; 31.0.15.5212) - Intel(R) Core(TM) i7-7700K CPU @ 4.20GHz (8 Threads)

### Issue description

While using the AnimationPlayer to change the pitch scale property of an AudioStreamPlayer, the pitch is unaltered, both in editor and at runtime.

### Steps to reproduce

Play an animation that changes the pitch scale property of an AudioStreamPlayer node and plays an AudioStream res.

### Minimal reproduction project (MRP)

[pitch_bug_test_project.zip](https://github.com/godotengine/godot/files/15215231/pitch_bug_test_project.zip)

### PR DESCRIPTION
**Pull request by killzebug**
Makes audio in animations use the pitch and volume properties of the AudioStreamPlayer they're played from.
Fixes: #91605

### CODE REVIEW

**Review by TokageItLab — CHANGES_REQUESTED** [review]
This is not correct fix and the wrong way to go about it.

The problem is that the AudioStreamPlayer properties are being ignored in the playback within AudioStreamPolyphonic, so AudioStreamPolyphonic needs to be fixed, not AnimationMixer.

Then, since AudioTrack is played back by AudioStreamPolyphonic, we need to add a feature to AudioTrackKey to allow it to set volume and pitch for each of them individually. And Instead of passing AudioStreamPlayer properties, we need to pass them individually from the key.

Simply put, AudioStreamPlayer properties should be treated as properties of the master track, and AudioTrackKeys should have properties for individual streams; It may be divided into two PRs as "Fix AudioStreamPlayer doesn't affect pitch and volume property to AudioStreamPolyphonic (bugfix)" and "Add volume and pitch property to AudioTrack key (enhancement)".

**Comment by Repiteo** [comment]
Superseded by #110525
